components/brave_shields/common/BUILD.gn

```diff
@@ -8,7 +8,6 @@ import("//mojo/public/tools/bindings/mojom.gni")
 static_library("common") {
   sources = [
     "adblock_domain_resolver.cc",
-    "adblock_domain_resolver.h",
```

Comment:
the `rust_cxx` build template has a parameter called `native_header_deps`, which allows specification of targets containing header files required for a successful build. i thought i'd move the header file so that the adblock gn refers to a header file within the relevant directory.

https://github.com/brave/brave-core/blob/31ffdcc7669fab50fca942461f88573ae17f4be3/components/brave_shields/adblock/rs/BUILD.gn#L20-L33

Reply:
@DJAndries I think you've got this backwards. The cxx target needs to depend on a target for adblock_domain_resolver.h and adblock_domain_resolver.cc
The way to deal with that is to move those files out of this target into their own target (in components/brave_shields/common) and depend on that, not remove the header file from this target

Reply:
but also why is adblock_domain_resolver in common? It never gets used in any process except the browser process so it doesn't belong here in the first place

Reply:
I'm not sure why the cxx target needs to depend on non-header files, since the cxx target seems to call cxxbridge under the hood, which seems to only perform code gen.

Also, the documentation for our rust_cxx template states that `native_header_deps` should include targets that contain header files referenced by the cxx binding.

https://github.com/brave/brave-core/blob/1b1460614af229585535805b1d5a2dd3d7e79601/build/rust/rust_cxx.gni#L34-L42

skus also uses this pattern.

https://github.com/brave/brave-core/blob/1b1460614af229585535805b1d5a2dd3d7e79601/components/skus/browser/rs/cxx/BUILD.gn#L21-L39